src/Parser.hs

```diff
@@ -70,7 +70,11 @@ parseConstant :: Parser Expr
 parseConstant = (string "pi" >> return Pi) <|> (string "e" >> return E)
 
 parseNumber :: Parser Expr
-parseNumber = (Num . read) <$> many1 (digit <|> char '.' <|> oneOf "-+eE")
+parseNumber = do
+    num <- many1 (oneOf "0123456789.")
+    if length (filter (=='.') num) > 1 
+        then fail "Invalid number format"
+        else return $ Num (read num)
```

Comment:
Think for a second about the structure of a floating point number. How many dots does it contain? What goes before and after the dot? Then express it as a parser.